### PR TITLE
Support extracting code when initializing test data without field name

### LINKED ISSUE #78: Extracting code for the specific test case does not work even when it conforms to the requirements
Even though some exercise tests seems to conform to the spec described here https://github.com/exercism/go-test-runner#subtest-format-specification, the test code on the website includes all test cases instead only the one that is currently tested.
Example where it is not working: Card Tricks Exercise
image

Example where the test case extraction is working correctly: Cars Assemble
image

### PR DESCRIPTION
**Pull request by meoconbatu**
This commit only supports anonymous struct type.
To fully support for named struct type requires a lot of changes in the source code.

Fixes #78

### CODE REVIEW

**Review by junedev — APPROVED** [review]
Good catch regarding the missing field names! I did not notice this was the issue why it was not working as expected.
Thanks for the fix. We need to wait for for someone from Exercism staff to approve before it can be merged.

**Review by ErikSchierboom — APPROVED** [review]

**Pull request merged**
